File: movie_translator/stages/identify.py
"""Identify stage — extract media identity from video file."""

from ..context import PipelineContext
from ..identifier import identify_media
from ..logging import logger


class IdentifyStage:
    name = 'identify'

    def run(self, ctx: PipelineContext) -> PipelineContext:
        logger.info(f'Identifying: {ctx.video_path.name}')
        ctx.identity = identify_media(ctx.video_path)

        if ctx.identity is not None and hasattr(ctx.identity, 'media_type'):
            identity = ctx.identity
            with ctx.metrics.span('record_identity') as s:
                s.detail('title', identity.title)
                s.detail('media_type', identity.media_type)
                s.detail('is_anime', identity.is_anime)
                if identity.season is not None:
                    s.detail('season', identity.season)
                if identity.episode is not None:
                    s.detail('episode', identity.episode)

        return ctx

File: crates/mt-pipeline/src/stages/identify.rs
//! Identify stage — extract media identity from the video file.

use mt_core::PipelineContext;
use mt_discovery::identify_media;

use crate::error::Result;
use tracing::info;

/// Stage role name.
pub const NAME: &str = "identify";

/// Identify the media and store it on the context.
pub fn run(mut ctx: PipelineContext) -> Result<PipelineContext> {
    info!(
        "Identifying: {}",
        ctx.video_path
            .file_name()
            .map(|n| n.to_string_lossy().to_string())
            .unwrap_or_default()
    );
    ctx.identity = Some(identify_media(&ctx.video_path)?);
    Ok(ctx)
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn name() {
        assert_eq!(NAME, "identify");
    }
}
